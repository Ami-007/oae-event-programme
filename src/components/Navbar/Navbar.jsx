import React from "react";
import { NavLink } from "react-router-dom";
import AboutIcon from "../NavbarIcon/AboutIcon";
import ScheduleIcon from "../NavbarIcon/ScheduleIcon";
import HomeIcon from "../NavbarIcon/HomeIcon";
import "./Navbar.scss";

const Navbar = (props) => {
  const { eventId } = props;

  return (
    <nav className="navbar">
      <div className="navbar__content">
        <NavLink
          // style={({ isActive }) => ({ fill: isActive ? "black" : "#CACBCD" })}
          className={({ isActive }) => (isActive ? "active" : "inactive")}
          to={`/${eventId}/event-schedule`}
        >
          <ScheduleIcon />
          <span className="navbar__title">SCHEDULE</span>
        </NavLink>
      </div>
      <div className="navbar__content">
        <NavLink
          // style={({ isActive }) => ({ fill: isActive ? "black" : "#CACBCD" })}
          className={({ isActive }) => (isActive ? "active" : "inactive")}
          to={`/${eventId}/home`}
        >
          <HomeIcon />
          <span className="navbar__title">HOME</span>
        </NavLink>
      </div>

      <div className="navbar__content">
        <NavLink
          className={({ isActive }) => (isActive ? "active" : "inactive")}
          to={`/${eventId}/our-story`}
          //   style={({ isActive }) => ({ fill: isActive ? "black" : "#CACBCD" })}
        >
          {/* <img className="navbar__icon" src={AboutIcon} alt="about icon" /> */}
          <AboutIcon />
          <span className="navbar__title">ABOUT</span>
        </NavLink>
      </div>
    </nav>
  );
};

export default Navbar;
