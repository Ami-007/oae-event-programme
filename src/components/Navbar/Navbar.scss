@use "../../assets/sass/variables.scss" as *;
@use "../../assets/sass/typography.scss" as *;

.navbar {
  padding: 1.25rem 0 1.25rem 0;
  background-color: rgb(255, 251, 251);
  display: flex;
  position: fixed;
  bottom: 0;
  width: 100%;
  &__content {
    flex: 1;
    text-align: center;
  }

  &__title {
    font-size: 0.7rem;
    font-family: "Jost", sans-serif;
    font-weight: 700;
  }

  .navbar__nav-link {
    text-decoration: none;

    &--active {
      fill: rgb(0, 0, 0);
      color: black;
    }
    &--inactive {
      fill: #cacbcd;
      color: #cacbcd;
    }
  }
}
