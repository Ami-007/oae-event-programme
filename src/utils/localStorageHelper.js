export const getDarkTheme = () => {
  //   if (darkTheme == true) {
  //     localStorage.setItem("--background-color", darkTheme);
  //   } else {
  //     localStorage.setItem("theme", "light");
  //   }
  console.log(localStorage.getItem("darkMode"));

  // convert isDarkMode to a Boolean
  // return that boolean

  return true;
};

export const setDarkTheme = (isDarkMode) => {
  localStorage.setItem("darkMode", isDarkMode.toString());
};
