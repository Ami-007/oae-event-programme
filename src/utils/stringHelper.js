export const getFontType = (fontType) => {
  if (typeof fontType === "string") {
    fontType = parseInt(fontType);
  }
  if (fontType === 1) {
    return "modern";
  } else if (fontType === 2) {
    return "classic";
  } else {
    return "modern";
  }
};

export default getFontType;
