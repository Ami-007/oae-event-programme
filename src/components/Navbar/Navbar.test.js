
import { screen } from "@testing-library/react";
import Navbar from "./Navbar";

import { customRender } from "../../utils/testUtils";
// import userEvent from "@testing-library/user-event";
// import { container } from "react-router-dom";

test("The links are contain in the page", () => {
  customRender(<Navbar/>);
  // links is an array of three link components
  const links = screen.getAllByRole('link');    

  // check each of these links has the correct href value. e.g.'/','/event-schedule','/our-story'

  // check RTL docs for how to get the href value of the link (which is an <a> tag)
  expect(links[0]).toHaveAttribute("href", '/event-schedule');
  expect(links[1]).toHaveAttribute("href", '/');
  expect(links[2]).toHaveAttribute("href", '/our-story')
  // toHaveAttribute

  
})

