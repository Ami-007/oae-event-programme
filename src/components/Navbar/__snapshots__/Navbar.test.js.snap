// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page link 1`] = `
<div>
  <nav
    class="navbar"
  >
    <div
      class="navbar__content"
    >
      <a
        class="schedule-link"
        href="/event-schedule"
      >
        <img
          alt="schedule icon"
          class="navbar__icon"
          src="schedule.svg"
        />
      </a>
      <span
        class="navbar__title"
      >
        SCHEDULE
      </span>
    </div>
    <div
      class="navbar__content"
    >
      <a
        class="home-link"
        href="/"
      >
        <img
          alt="home icon"
          class="navbar__icon"
          src="home.svg"
        />
      </a>
      <span
        class="navbar__title"
      >
        HOME
      </span>
    </div>
    <div
      class="navbar__content"
    >
      <a
        class="about-link"
        href="/our-story"
      >
        <img
          alt="about icon"
          class="navbar__icon"
          src="info.svg"
        />
      </a>
      <span
        class="navbar__title"
      >
        ABOUT
      </span>
    </div>
  </nav>
</div>
`;
